#35 마일리지 조회 기능 추가

0hun opened the pull request:
나의 마일리지 조회 기능 추가
테이블 추가하여 쿼리 추가

0hun committed:
#35 마일리지 조회 기능 추가

나의 마일리지 조회 기능 추가
테이블 추가하여 쿼리 추가

0hun committed:
#35 마일리지 상태 값 추가

마일리지 상태 값 추가
마일리지 삭제되지 않는 마일리지만 조회하도록 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/mileageController.java:
```diff
@@ -0,0 +1,48 @@
+   * @return ResponseEntity
+   */
+  @LoginCheck(userLevel = UserLevel.USER)
+  @GetMapping("/my")
```
이 정도 된다면 마일리지를 포함한 자신의 정보들을 관리하는쪽을 별도의 Controller로 분리하는게 어떨까요?

f-lab-dev commented on src/main/java/com/younghun/hibusgo/domain/Mileage.java:
```diff
@@ -0,0 +1,25 @@
+@NoArgsConstructor
+@Getter
+@Builder
+@ToString
```
이거는 왜 넣어주셨을까요?

0hun replied:
마일리지를 계산하고 저장할때 캐시에 저장하기 위해 직렬화가 필요하므로 추가했는데, 현재는 필요없으므로 지우도록하겠습니다.

0hun committed:
#35 리뷰 반영

관리자가 특정 회원의 마일리지를 조회하는 메소드 추가
회원 자신의 마일리지를 조회하는 메소드 추가

0hun committed:
Merge branch 'develop' into issue/35

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/AccountController.java:
```diff
@@ -146,4 +153,21 @@ public void passwordInitBinder(WebDataBinder webDataBinder) {
+     * @return ResponseEntity
+     */
+    @LoginCheck(userLevel = UserLevel.USER)
+    @GetMapping("/myMileage")
```
이 url은 rest 규약에 맞을까요?

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/AccountController.java:
```diff
@@ -146,4 +153,21 @@ public void passwordInitBinder(WebDataBinder webDataBinder) {
+     */
+    @LoginCheck(userLevel = UserLevel.USER)
+    @GetMapping("/myMileage")
+    public ResponseEntity<?> getMyMileage(@SessionId long accountId) {
```
`accountId`와 `sessionId` 같은 뜻일까요? 어노테이션 이름을 바꿔주는게 좋겠네요~

0hun replied:
넵 알겠습니다.

0hun committed:
#35 마일리지 조회 기능 추가

sessionId가 로그인한 계정 accountId와 같은 의미이므로 혼동되지않게 accountId로 변경

로그인한 유저의 마일리지 조회 url 변경

0hun committed:
Merge branch 'issue/35' of https://github.com/f-lab-edu/hi-bus-go into issue/35

0hun committed:
Merge branch 'develop' into issue/35

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/AccountController.java:
```diff
@@ -100,7 +100,7 @@ public void passwordInitBinder(WebDataBinder webDataBinder) {
     @LoginCheck(userLevel = UserLevel.USER)
     @DeleteMapping("/myInfo")
-    public ResponseEntity<?> deleteAccount(@SessionId Long accountId) {
+    public ResponseEntity<?> deleteAccount(@AccountId Long accountId) {
```
현재 로그인한 유저의 ID라는 뜻을 더 잘 전달할 수 있는 어노테이션 이름이 있을지 한번 고민해주세요~
만약 떠오르지 않는다면 그냥 머지하셔도 상관은 없습니다
